### PR TITLE
mod_dav: add DavBasePath

### PR DESCRIPTION
**Pull request by notroj**

### CODE REVIEW

File: docs/manual/mod/mod_dav.xml [hunk 1]
```diff
@@ -194,6 +194,36 @@ Alias "/php-source" "/home/gstein/php_files"
 </usage>
 </directivesynopsis>
 
+<directivesynopsis>
+<name>DavBasePath</name>
+<description>Configure repository root path</description>
+<syntax>DavBasePath <var>root-path</var></syntax>
+<default>DavBasePath /repos/</default>
```
Comment (tititiou36) [inline docs/manual/mod/mod_dav.xml]: Is "/repos/" as the default value really meaningful?

It is not hard-coded in the module and does not fit well with the comment "Take the repos root from DAVBasePath if configured, **else the path of the enclosing section**" in the [cb4eb29](https://github.com/apache/httpd/pull/377/commits/cb4eb2940eaab5103d9055e2fe25e5a3e717a132)

File: modules/dav/main/mod_dav.c [hunk 2]
```diff
@@ -765,11 +779,27 @@ DAV_DECLARE(dav_error *) dav_get_resource(request_rec *r, int label_allowed,
...
-    err = (*conf->provider->repos->get_resource)(r, conf->dir,
+    err = (*conf->provider->repos->get_resource)(r, base,
                                                  label, use_checked_in,
                                                  res_p);
     if (err != NULL) {
+        /* In the error path, give a hint that DavBasePath needs to be
+         * used if the location was configured via a regex match. */
+        if (!conf->base) {
+            core_dir_config *conf = ap_get_core_module_config(r->per_dir_config);
+
```
Comment (rpluem) [inline modules/dav/main/mod_dav.c:795]: Is it a good idea to use the same name `conf`for the `core_dir_config` variable as for the `dav_dir_conf` variable and thus mask it in the block?

File: docs/manual/mod/mod_dav.xml (hunk 1 above)
Comment (rpluem) [inline docs/manual/mod/mod_dav.xml]: Maybe

```suggestion
<default>None</default>
```

File: modules/dav/main/mod_dav.c (hunk 2 above)
Comment (notroj) [reply modules/dav/main/mod_dav.c:795]: Definitely a bad idea! Fixed in 706af916933612750c1cf8f165e23d37c9b865b2

File: docs/manual/mod/mod_dav.xml (hunk 1 above)
Comment (notroj) [reply docs/manual/mod/mod_dav.xml]: Thanks for catching this, fixed in 59c725a

Comment (notroj) [reply docs/manual/mod/mod_dav.xml]: Nope, definitely wouldn't be... thanks for reviewing.

**Comment by notroj** [comment]
Thanks a lot for the reviews on this PR - updated now.

**Comment by notroj** [comment]
Merged in ddd173bc63477cbbf45586f68c951e09659c7c4e

**Pull request closed without merging**